runelite-client/src/main/java/net/runelite/client/plugins/twitch/TwitchPlugin.java

```diff
@@ -154,6 +187,228 @@ public void onConfigChanged(ConfigChanged configChanged)
 		connect();
 	}
 
+	@Subscribe
+	public void onSetMessage(SetMessage setMessage)
+	{
+		if (client.getGameState() != GameState.LOADING && client.getGameState() != GameState.LOGGED_IN)
+		{
+			return;
+		}
+
+		if (setMessage.getType() == ChatMessageType.TWITCH && setMessage.getMessageNode().getSender().contains("Twitch"))
+		{
+			beautifyTwitchMessage(setMessage);
+		}
+	}
+
+	private void beautifyTwitchMessage(final SetMessage message)
+	{
+		StringBuilder sb = new StringBuilder();
+
+		//System.out.println(message.getMessageNode().getRuneLiteFormatMessage());
+
+		// TODO: Make the Sender (Twitch) text purple
+		String senderInfo = constructBadgeIcons(message.getMessageNode().getName());
+
+		sb.append(message.getMessageNode().getValue().replaceAll("<colNORMAL>", ColorUtil.colorTag(chatColor)));
+		//sb.insert(sb.length(), "</col>");
+
+		message.getMessageNode().setSender("<col=8e00ba>" + message.getMessageNode().getSender() + "</col>");
+		message.getMessageNode().setName(senderInfo);
+		message.getMessageNode().setValue(sb.toString());
+
+		//client.refreshChat();
+	}
+
+	private void repaintChat()
+	{
+		// All messages
+		Widget messages = client.getWidget(162, 58);
```

Comment:
Use `WidgetInfo`, which will remove the need to comment what this widget is meant to be.